scala/src/main/scala/smile/clustering/k-modes/KModes.scala

```diff
@@ -0,0 +1,63 @@
+package smile.clustering
+
+import scala.collection.mutable
+import scala.util.Random
+import smile.clustering.distances.Binary_Distance
+
+class KModes(data: Array[(Int, Array[Int])], k: Int, epsilon: Double, jmax: Int, metric: Binary_Distance) extends ScalaClusteringTypes
```

Comment:
Please add documents and references to the algorithms. Actually, please add documents to all your classes. Thanks.